src/meta_ops.rs

```diff
@@ -396,3 +400,222 @@ pub fn equal<'gc>(
+            .or(Err(()))?,
+
+        (Value::Table(a), Value::Table(b)) => {
+            let get_sub = |t: Table<'gc>| {
+                let eq = t.metatable().map(|t| t.get(ctx, m)).unwrap_or_default();
+                if eq.is_nil() {
+                    None
+                } else {
+                    Some(eq)
+                }
+            };
+            if let Some(a_eq) = get_sub(a) {
+                MetaResult::Call(MetaCall {
+                    function: call(ctx, a_eq).or(Err(()))?,
+                    args: [a.into(), b.into()],
+                })
+            } else if let Some(b_eq) = get_sub(b) {
+                MetaResult::Call(MetaCall {
+                    function: call(ctx, b_eq).or(Err(()))?,
+                    args: [a.into(), b.into()],
+                })
+            } else {
+                Err(())?
+            }
+        }
+        (Value::UserData(a), Value::UserData(b)) => {
+            let get_sub = |t| get_metamethod_from_userdata(ctx, t, m);
+            if let Some(a_eq) = get_sub(a) {
+                MetaResult::Call(MetaCall {
+                    function: call(ctx, a_eq).or(Err(()))?,
+                    args: [a.into(), b.into()],
+                })
+            } else if let Some(b_eq) = get_sub(b) {
+                MetaResult::Call(MetaCall {
+                    function: call(ctx, b_eq).or(Err(()))?,
+                    args: [a.into(), b.into()],
+                })
+            } else {
+                Err(())?
+            }
+        }
+        (Value::UserData(a), Value::Table(b)) => {
+            if let Some(a_eq) = get_metamethod_from_userdata(ctx, a, m) {
+                MetaResult::Call(MetaCall {
+                    function: call(ctx, a_eq).or(Err(()))?,
+                    args: [a.into(), b.into()],
+                })
+            } else if let Some(b_eq) = get_metamethod_from_table(ctx, b, m) {
+                MetaResult::Call(MetaCall {
+                    function: call(ctx, b_eq).or(Err(()))?,
+                    args: [a.into(), b.into()],
+                })
+            } else {
+                Err(())?
+            }
+        }
+        (Value::Table(a), Value::UserData(b)) => {
+            if let Some(a_eq) = get_metamethod_from_table(ctx, a, m) {
+                MetaResult::Call(MetaCall {
+                    function: call(ctx, a_eq).or(Err(()))?,
+                    args: [a.into(), b.into()],
+                })
+            } else if let Some(b_eq) = get_metamethod_from_userdata(ctx, b, m) {
+                MetaResult::Call(MetaCall {
+                    function: call(ctx, b_eq).or(Err(()))?,
+                    args: [a.into(), b.into()],
+                })
+            } else {
+                Err(())?
+            }
+        }
+
+        (Value::Table(t), a) => {
+            let get_sub = |t: Table<'gc>| {
+                let eq = t
+                    .metatable()
+                    .map(|t| t.get(ctx, MetaMethod::Sub))
```

Comment:
I think there are several places where MetaMethod::Sub is still used instead of the metamethod parameter?

Reply:
Fixed now. I've condensed the code down further so there shouldn't be any more cases.